account_check_deposit/models/account_check_deposit.py

```diff
@@ -0,0 +1,368 @@
+    move_id = fields.Many2one(
+        comodel_name="account.move",
+        string="Journal Entry",
+        readonly=True,
+        check_company=True,
+    )
+    bank_journal_id = fields.Many2one(
+        comodel_name="account.journal",
+        string="Bank Account",
+        required=True,
+        domain="[('company_id', '=', company_id), ('type', '=', 'bank'), "
+        "('bank_account_id', '!=', False)]",
+        check_company=True,
+        states={"done": [("readonly", "=", True)]},
+        tracking=True,
+    )
+    line_ids = fields.One2many(
+        comodel_name="account.move.line",
+        related="move_id.line_ids",
+        string="Lines",
+    )
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        required=True,
+        states={"done": [("readonly", "=", True)]},
+        default=lambda self: self.env.company,
+        tracking=True,
+    )
+    total_amount = fields.Monetary(
+        compute="_compute_check_deposit",
+        store=True,
+        currency_field="currency_id",
+        tracking=True,
+    )
+    check_count = fields.Integer(
+        compute="_compute_check_deposit",
+        store=True,
+        string="Number of Checks",
+        tracking=True,
+    )
+    is_reconcile = fields.Boolean(
+        compute="_compute_check_deposit", store=True, string="Reconcile"
+    )
+
+    _sql_constraints = [
+        (
+            "name_company_unique",
+            "unique(company_id, name)",
+            "A check deposit with this reference already exists in this company.",
+        )
+    ]
+
+    @api.depends(
+        "company_id",
+        "currency_id",
+        "check_payment_ids.debit",
+        "check_payment_ids.amount_currency",
+        "move_id.line_ids.reconciled",
+    )
+    def _compute_check_deposit(self):
+        rg_res = self.env["account.move.line"].read_group(
+            [("check_deposit_id", "in", self.ids)],
+            ["check_deposit_id", "debit", "amount_currency"],
+            ["check_deposit_id"],
+        )
+        mapped_data = {
+            x["check_deposit_id"][0]: {
+                "debit": x["debit"],
+                "amount_currency": x["amount_currency"],
+                "count": x["check_deposit_id_count"],
+            }
+            for x in rg_res
+        }
+
+        for deposit in self:
+            reconcile = False
+            company_cur = deposit.company_id.currency_id
+            if company_cur != deposit.currency_id:
+                total = mapped_data.get(deposit.id, {"amount_currency": 0.0})[
+                    "amount_currency"
+                ]
+            else:
+                total = mapped_data.get(deposit.id, {"debit": 0.0})["debit"]
+            count = mapped_data.get(deposit.id, {"count": 0})["count"]
+            if deposit.move_id:
+                for line in deposit.move_id.line_ids:
+                    if not company_cur.is_zero(line.debit) and line.reconciled:
+                        reconcile = True
```

Comment:
Myself note:
* can we have more than one line with debit > 0 ? I don't think so
* why use the last line state only ? because we expect to get one record

Should be the same with sugar syntax
```suggestion
            if deposit.move_id.line_ids.filtered(lambda line: not company_cur.is_zero(line.debit) and line.reconciled):
                reconcile = True
```